docs/en/modules/net_info.md

```diff
@@ -0,0 +1,79 @@
+# net_info Module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2017-01-21 | [smcl](http://blog.mclemon.io/esp8266-contributing-to-the-nodemcu-ecosystem) | [wolfgangr](https://github.com/wolfgangr/nodemcu-firmware) | [net_info.c](../../../app/modules/net_info.c)
+
+
+This module is a stub to collect common network diagnostic and analysis tools.
+
+##net_info.ping()
+ send ICMP ECHO_REQUEST to network hosts 
+ 
+**Synopsis:**:
+```
+net_info.ping(host [, count [, callback]]) 
```

Comment:
hmm....
as I statet, I have just portet code found somewhere else.
I get a rough grasp of your points, but can't follow them in detail.
Getting familiar with the details of C/LUA interfacing is still on my todo list.
But time is limited, and my priority is to have an ethernet driver running.
I'd suppose to postpone this issue until I have to learn the details of lua interfacing anyway.

Another option is to ask the original author of the code, @smcl, to take over.
As I see, he has just dropped into the discussin on topic #1758
Is there a formal way to get him on board at this stage of a pull request?

A third option would be to remove the callback functionality, since this requires most of the lua / stack interfacing. But this causes work, test and risk as well.
Personally, I'd prefer to fix things than to dispose them.